Wrappers/Python/cil/optimisation/functions/TotalVariation.py

```diff
@@ -294,43 +331,45 @@ def _fista_on_dual_rof(self, x, tau, out = None):
             out = self.gradient.domain_geometry().allocate(0)   
 
         should_break = False
-        for k in range(self.iterations):
+        for k in range(self.iterations): # line 3 in alogirhtm one of "Multicontrast MRI Reconstruction with Structure-Guided Total Variation", Ehrhardt, Betcke, 2016.
                                                                                    
             t0 = t
-            self.gradient.adjoint(tmp_q, out = out)
-            out.sapyb(tau_reg_neg, x, 1.0, out=out)
-            self.projection_C(out, tau=None, out = out)
+            self.gradient.adjoint(tmp_q, out = out) # line 4
+            out.sapyb(tau_reg_neg, x, 1.0, out=out)# line 4
+            self.projection_C(out, tau=None, out = out)# line 4 
 
             if should_break:
                 break
 
-            self.gradient.direct(out, out=p1)
+            self.gradient.direct(out, out=p1) # line 4
 
-            multip = (-self.L)/tau_reg_neg
-            p1.multiply(multip,out=p1)
+            multip = (-self.L)/tau_reg_neg# line 4
+            p1.multiply(multip,out=p1) #line 4/5 
             
-            tmp_q += p1
+            tmp_q += p1 # line 5
 
-            if self.tolerance is not None and k%5==0:
+            if self.tolerance is not None and k%5==0: # testing convergence criterion 
                 error = p1.norm()
```

Comment:
If you implement the change above with `sapyb` you need to change the way `error` is calculated because `p1` is not multiplied by `multip`.

```suggestion
                error = p1.norm() * multip
```

Reply:
Done - thanks Edo